compiler/AST/AstToText.cpp

```diff
@@ -1309,6 +1417,107 @@ void AstToText::appendExpr(NamedExpr* expr, bool printingType)
   appendExpr(expr->actual, printingType);
 }
 
+void AstToText::appendExpr(IfExpr* expr, bool printingType)
+{
+  mText += "if ";
+  if (Expr* sel = toExpr(expr->getCondition()))
+  {
+    appendExpr(sel, printingType);
+
+    mText += " then ";
+    if (BlockStmt* thenBlockStmt = toBlockStmt(expr->getThenStmt()))
+    {
+      if (Expr* exp = toExpr(thenBlockStmt->body.get(1)))
+      {
+        appendExpr(exp, printingType);
+
+        mText += " else ";
+        if (BlockStmt* elseBlockStmt = toBlockStmt(expr->getElseStmt()))
+        {
+          if (Expr* exp = toExpr(elseBlockStmt->body.get(1)))
```

Comment:
Sorry, I wasn't very clear.  The code I was worried could get dropped on the floor was if the BlockStmt's body had more than one expression in it - you're doing `body.get(1)` here, but not checking `body.length` to ensure it is only 1.

Reply:
Ok. I thought there should only be one statement. So, should I have it for more than one statement too?

Reply:
I doubt it will be valid code in that case, so I think just asserting that the length matches 1 is sufficient